### PR TITLE
Makes Dredd a monorepo

### PR DESCRIPTION
**Pull request by artem-zakharchenko**
#### :rocket: Why this change?

- To manage Dredd and friends more effectively

#### :memo: Related issues and Pull Requests

- A part of #1186 

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

**Comment by artem-zakharchenko** [comment]
**Current state:** Managing docs, moved to the root directory, fixing references to fixtures from `hooks/index.rst`. There were some issues with `docutils` python dependency being not installed. Should be solved with moving docs to the root.

File: packages/dredd/package.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    "prettify": "prettier --write lib/**/*.js test/**/*.js",
+    "prettify:check": "prettier --check lib/**/*.js test/**/*.js",
+    "test": "mocha \"./test/**/*-test.js\"",
+    "test:debug": "mocha --debug-brk \"./test/**/*-test.js\"",
+    "e2e": "npm run e2e:apib && npm run e2e:openapi2",
+    "e2e:apib": "cucumber-js",
+    "e2e:openapi2": "cucumber-js --world-parameters=\"{\\\"apiDescriptionFormat\\\": \\\"application/swagger+yaml\\\"}\"",
+    "ci:lint": "npm run lint",
+    "ci:test": "npm test",
+    "ci:smoke": "bash ./scripts/smoke.sh",
```
Comment (kylef) [inline packages/dredd/package.json]: I don't think this comment is about the PR itself, but once we have dredd-transactions in the repo too, we will want to alter the smoke tests to test with the in-repo dredd-transactions.

When the time comes for this, you can take a look at how I solved similiar in API Elements where "Fury CLI" uses the local in-repo libraries as part of the smoke test: https://github.com/apiaryio/api-elements.js/blob/dbd6aafbd50fcb2fab43eae9d612561147308377/.circleci/smoke.sh

**Comment by pksunkara** [comment]
LGTM.

**Comment by artem-zakharchenko** [comment]
@honzajavorek please note that this pull request also brings Yarn as the default package manager to Dredd. The reasoning is the support for "workspaces", which grants an ability to properly install shared dependencies between multiple packages in a monorepo.

Due to that, I'd also update the docs (mainly contributing guidelines) to mention the Yarn preference. What do you think?

**Comment by honzajavorek** [comment]
@artem-zakharchenko 

> Yarn as the default package manager to Dredd

For development, right? So we're talking about the "internals" page in the docs. Sure, reasonable.

For Dredd users we should still recommend npm.

**Comment by artem-zakharchenko** [comment]
Yes, of course. Publicly nothing changes, Yarn is preferred for contributing only, not to fix package managers during dependencies management and CI. Well, in that case you're right, nothing to update (except of contributing docs in a separate pull request).

File: appveyor.yml [hunk 2]
```diff
@@ -3,14 +3,16 @@ environment:
 install:
   - ps: Install-Product node 12
   - 'npm -g install npm@6'
```
Comment (honzajavorek) [inline appveyor.yml]: I guess we can remove this line now

File: docs/conf.py [hunk 3]
```diff
@@ -116,6 +116,9 @@
 
 linkcheck_ignore = [
     'https://crates.io/crates/dredd-hooks',  # https://github.com/sphinx-doc/sphinx/pull/5140
+    # Exclude Dredd repository links, because they are checked
+    # as a part of another script.
```
Comment (honzajavorek) [inline docs/conf.py]: Would be nice to mention which one - `_extensions/ghlink_check.py`

File: appveyor.yml (hunk 2 above)
Comment (artem-zakharchenko) [reply appveyor.yml]: Removed in https://github.com/apiaryio/dredd/pull/1531/commits/9abe641a834d0bb9609dae2e2df8c38bd106d8f2.

File: docs/conf.py (hunk 3 above)
Comment (artem-zakharchenko) [reply docs/conf.py]: Mentioned in https://github.com/apiaryio/dredd/pull/1531/commits/5cc3afda02db6774f0256bb5de262335528f5905.

**Comment by artem-zakharchenko** [comment]
Experiencing a flaky test (#1519). Re-running the workflow...

**Pull request merged**